dmoj/cptbox/isolate.py

```diff
@@ -56,22 +57,28 @@ def __init__(self, read_fs, write_fs=None, writable=(1, 2)):
         self.update(
             {
                 # Deny with report
-                sys_openat: self.check_file_access_at('openat', FilesystemSyscallKind.OPEN),
-                sys_open: self.check_file_access('open', 0, FilesystemSyscallKind.OPEN),
-                sys_faccessat: self.check_file_access_at('faccessat', FilesystemSyscallKind.READ),
-                sys_faccessat2: self.check_file_access_at('faccessat2', FilesystemSyscallKind.READ),
-                sys_access: self.check_file_access('access', 0, FilesystemSyscallKind.READ),
-                sys_readlink: self.check_file_access('readlink', 0, FilesystemSyscallKind.READ),
-                sys_readlinkat: self.check_file_access_at('readlinkat', FilesystemSyscallKind.READ),
-                sys_stat: self.check_file_access('stat', 0, FilesystemSyscallKind.READ),
-                sys_stat64: self.check_file_access('stat64', 0, FilesystemSyscallKind.READ),
-                sys_lstat: self.check_file_access('lstat', 0, FilesystemSyscallKind.READ),
-                sys_lstat64: self.check_file_access('lstat64', 0, FilesystemSyscallKind.READ),
-                sys_fstatat: self.check_file_access_at('fstatat', FilesystemSyscallKind.FSTAT),
-                sys_statx: self.check_file_access_at('statx', FilesystemSyscallKind.FSTAT),
-                sys_tgkill: self.do_kill,
-                sys_kill: self.do_kill,
-                sys_prctl: self.do_prctl,
+                sys_openat: self.handle_openat('openat', dir_reg=0, file_reg=1, flag_reg=2),
+                sys_open: self.handle_open('open', file_reg=0, flag_reg=1),
+                sys_faccessat: self.handle_file_access_at(
+                    'faccessat', FilesystemSyscallKind.READ, dir_reg=0, file_reg=1
+                ),
+                sys_faccessat2: self.handle_file_access_at(
+                    'faccessat', FilesystemSyscallKind.READ, dir_reg=0, file_reg=1
```

Comment:
Kind of `faccessat`? But it's `faccessat2`.

Reply:
Lol, fixed.